### PR TITLE
license: intial spdx

### LINKED ISSUE #1458: [FEATURE] Switch to SPDX license identifiers
What is your take on switching to SPDX identifiers [1]? The license text at the beginning of each file makes code harder to read. 

[1] https://spdx.org/ids-how

### LINKED ISSUE #1497: [BUG] Coefficients files need license info
Files below are missing licenses. Please add BSD license + copyright. Actually these files without that shouldn't be even in repo. Blocker for #1458 

Sorry for troubling you, but author should apply that initial license. I don't know if I can touch these files without your permission.

List of files:
```
./sof/src/include/sof/audio/coefficients/pdm_decim/pdm_decim_int32_03_4375_5100_010_095.h
./sof/src/include/sof/audio/coefficients/pdm_decim/pdm_decim_int32_03_3850_5100_010_095.h
./sof/src/include/sof/audio/coefficients/pdm_decim/pdm_decim_int32_02_4288_5100_010_095.h
./sof/src/include/sof/audio/coefficients/pdm_decim/pdm_decim_int32_06_4156_5100_010_095.h
./sof/src/include/sof/audio/coefficients/pdm_decim/pdm_decim_int32_02_4375_5100_010_095.h
./sof/src/include/sof/audio/coefficients/pdm_decim/pdm_decim_int32_08_4156_5380_010_090.h
./sof/src/include/sof/audio/coefficients/pdm_decim/pdm_decim_int32_04_4375_5100_010_095.h
./sof/src/include/sof/audio/coefficients/pdm_decim/pdm_decim_int32_05_4331_5100_010_095.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_3_1_2268_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_5_4_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_8_7_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_4_5_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_20_21_4167_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_10_21_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_32_21_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_4_3_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_3_1_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_21_20_1667_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_7_8_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_3_4_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_40_21_3968_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_21_80_3968_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_3_2_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_20_7_2976_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_6_5_4354_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_21_40_3968_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_5_6_4354_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_table.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_2_1_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_3_2_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_2_1_2268_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_1_2_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_1_3_2268_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_table.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_define.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_1_2_2268_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_8_7_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_1_3_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_25_24_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_8_7_2468_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_24_25_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_6_1_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_2_3_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_3_1_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_5_7_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_7_8_4535_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_20_21_1667_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_1_6_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_21_20_4167_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_define.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_2_3_1814_5000.h
./sof/src/include/sof/audio/coefficients/src/src_std_int32_8_21_3239_5000.h
./sof/src/include/sof/audio/coefficients/src/src_tiny_int16_1_3_1814_5000.h

```

### LINKED ISSUE #1499: [BUG] GDB files need license info
Files below are missing licenses. Please add BSD license + copyright. Actually these files without that shouldn't be even in repo. Blocker for #1458 

Sorry for troubling you, but author should apply that initial license. I don't know if I can touch these files without your permission.

List of files:
```
./sof/src/gdb/ringbuffer.c
./sof/src/include/sof/gdb/ringbuffer.h
./sof/src/include/sof/gdb/gdb.h

```

### LINKED ISSUE #1500: [BUG] KPB UT files need license info
Files below are missing licenses. Please add BSD license + copyright. Actually these files without that shouldn't be even in repo. Blocker for #1458 

Sorry for troubling you, but author should apply that initial license. I don't know if I can touch these files without your permission.

List of files:
```
./sof/test/cmocka/src/audio/kpb/kpb_mock.h
./sof/test/cmocka/src/audio/kpb/kpb_mock.c

```

### PR DESCRIPTION
**Pull request by jajanusz**
As discussed in #1458 

It covers all our FW sources (c/h/S) - 3rd party files (f.e. from Tensilica) are not changed.
It add identifiers to Perl/Python/Shell scripts and Kconfig files.

I don't know if m4 and Matlab should have them. Should I add them? It's not a big deal, I'm just not sure if we have copyrights to all of them.

I used C-style comments in .S files, however they are mixed in kernel repo, so look like there is no clear convention.

This PR changes license of rimage from GPL to BSD, so @dbaluta needs to agree with that change as he is also author in files that were GPL-licensed.

I added missing licenses on behalf of Intel employees.

This closes #1458, closes #1497, closes #1499, closes #1500

### CODE REVIEW

**Comment by jajanusz** [comment]
@lgirdwood 
These files:
https://github.com/thesofproject/sof/blob/master/src/platform/baytrail/include/arch/xtensa/config/core-isa.h
https://github.com/thesofproject/sof/blob/master/src/platform/haswell/include/arch/xtensa/config/core-isa.h

Have no license. Are they made by Intel as wrapper to -hsw/-bdw files form Tensilica or just tensilica files? I don't know how their `Copyright` notice should look without that info.

**Comment by lgirdwood** [comment]
@jajanusz they should have same copyright as APL versions, both derived from same place and generated by Intel.

File: rimage/cse.c [hunk 1]
```diff
@@ -1,18 +1,9 @@
...
- *
- * This program is distributed in the hope it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
- * more details.
- *
- *  Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
- *          Keyon Jie <yang.jie@linux.intel.com>
- */
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (plbossart) [inline rimage/cse.c]: Why are we changing a license? What is the reason why this was GPL in the first place? 
To me the change of license should be a separate PR that's well tracked, not hidden in a script-based code reshuffling.

Comment (jajanusz) [reply rimage/cse.c]: It's not hidden. I explicitly mentioned it in PR description and wrote who needs to agree.

Comment (plbossart) [reply rimage/cse.c]: well the net result is that we merged a PR that changes the license without explicit agreement from one of the authors.

Comment (lgirdwood) [reply rimage/cse.c]: @plbossart original GPL was an error (I probably copy and pasted the wrong licence from somewhere), it should be BSD.

Comment (jajanusz) [reply rimage/cse.c]: @plbossart @lgirdwood Even if it was mistake, the people that was adding code to these files were adding them under GPL license (because in the moment of contribution there was GPL license), so you still need consent from them imho. Just ask @dbaluta for agreement to be safe. He can do it now even if it's already merged, and if he disagrees we can revert this PR.

Comment (lgirdwood) [reply rimage/cse.c]: @dbaluta let us know if you disagree with BSD for rimage and we can revert that part.

**Comment by dbaluta** [comment]
I agree with the license change. Sorry for the late answer, I was in vacation :).
